src/ant/include/ant/AntennaChecker.hh

```diff
@@ -68,6 +68,48 @@ struct Violation
   int diode_count_per_gate;
 };
 
+// A struct that contains a net as well as one of its routing levels.
+// This struct is used to index the map, allowed_wire_length which is created 
+// in the class below.
+struct NetLayerPair
+{
+    dbNet* net;
+    int routing_level;
+ 
+    // default constructor
+    NetLayerPair() {}
+    // constructor
+    NetLayerPair(dbNet* net, int routing_level): net(net), routing_level(routing_level) {}
```

Comment:
Why not default constructor?

Reply:
Why not typedef std::pair<dbNet*, int>?

Reply:
@jjcherry56 initially I had the NetLayerPaid and the MaxLength as std::pair<dbNet*, int> and std::pair<double, double> respectively but @maliberty recommended using Structs instead so that they have meaningful names. Should I convert them back to Pairs and use typedef to provide the names?

Reply:
Since it is only used as a key for allowed_wire_length it is overkill IMHO

Reply:
I never suggest using pair as first/second are meaningless member names.